pkg/e2e/verify/encrypted_storage.go

```diff
@@ -0,0 +1,354 @@
+			serviceAccountJson, err := createGCPServiceAccount(h, testInstanceName, h.CurrentProject())
+			Expect(err).ToNot(HaveOccurred(), "Error creating credentialsrequest")
+
+			testKey, err := createGCPKey(h, serviceAccountJson, testInstanceName)
+			Expect(err).ToNot(HaveOccurred(), "Error creating encryption key in GCP KMS.")
+
+			ginkgo.By("Using the key to create an encrypted an encrypted storageclass as a dedicated-admin")
+			// RBAC rules must be set manually if not testing with a ccs cluster
+			// (gcp dedicated-admins only have permissions to create/modify storageclasses on ccs clusters)
+			if !viper.GetBool(ocmprovider.CCS) {
+				_, err := h.Kube().RbacV1().ClusterRoles().Create(context.TODO(), &rbacv1.ClusterRole{
+					TypeMeta: metav1.TypeMeta{
+						Kind: "ClusterRole",
+						APIVersion: "rbac.authorization.k8s.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testInstanceName,
+						Labels: map[string]string {
+							"managed.openshift.io/aggregate-to-dedicated-admins": "cluster",
+						},
+					},
+					Rules: []rbacv1.PolicyRule{{
+						Verbs: []string{"*"},
+						APIGroups: []string{"storage.k8s.io"},
+						Resources: []string{"storageclasses"},
+					}},
+				}, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred(), "Error creating clusterrole '" + testInstanceName + "':")
+			}
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "dummy-admin@redhat.com",
+				Groups: []string{
+					"dedicated-admins",
+				},
+			})
+
+			// Updated RBAC rules may take time to apply -> poll until we can successfully create a storageclass
+			volumeBindingMode := storagev1.VolumeBindingWaitForFirstConsumer
+			wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+				_, err = h.Kube().StorageV1().StorageClasses().Create(context.TODO(), &storagev1.StorageClass{
+					TypeMeta: metav1.TypeMeta{
+						Kind: "StorageClass",
+						APIVersion: "storage.k8s.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testInstanceName,
+					},
+					Provisioner: "pd.csi.storage.gke.io",
+					Parameters: map[string]string {
+						"type": "pd-standard",
+						"disk-encryption-kms-key": testKey,
+					},
+					VolumeBindingMode: &volumeBindingMode,
+				}, metav1.CreateOptions{})
+				if err != nil { return false, nil }
+				return true, err
+			})
+			Expect(err).ToNot(HaveOccurred(), "Error creating storageclass '" + testInstanceName + "':")
+
+			ginkgo.By("Verifying the storageclass creates encrypted PVCs")
+			// Ensure pods can read/write on encrypted PVs
+			pvc, err := h.Kube().CoreV1().PersistentVolumeClaims(h.CurrentProject()).Create(context.TODO(), &corev1.PersistentVolumeClaim{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "PersistentVolumeClaim",
+					APIVersion: "core/v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: testInstanceName,
+					Namespace: h.CurrentProject(),
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
```

Comment:
Am I missing something? It doesn't seem like you are passing a CMK to this StorageClass...

Reply:
The key is passed to the storageclass here https://github.com/openshift/osde2e/blob/76e95160d422f6d200ef73d784c06009405ae4f9/pkg/e2e/verify/encrypted_storage.go#L112, and verified to be used to encrypt the disk here https://github.com/openshift/osde2e/blob/76e95160d422f6d200ef73d784c06009405ae4f9/pkg/e2e/verify/encrypted_storage.go#L203 (I figured it'd be a bit out of the scope of this test to verify that Google is telling the truth when it claims a PV is encrypted with a specific key :) ).